src/berry_crush_3.c

```diff
@@ -830,513 +831,74 @@ static void sub_814DF44(u8 windowId, u8 left, u8 colorId, const u8 *string)
     AddTextPrinterParameterized3(windowId, 2, left, 0, sBerryCrushTextColorTable[colorId], 0, string);
 }
 
-#ifdef NONMATCHING
 static void sub_814DFA4(struct BerryCrushGame * sp0C, u8 sp10, u8 sp14, u8 sp18)
 {
-    u8 r8;
-    u8 sp1C;
-    u8 sp20;
+    u8 r8 = 0;
+    u8 sp1C = 0;
+    u8 sp20 = 0;
     u8 r2;
     s32 r3;
     u8 r7;
-    struct BerryCrushGame_68_x * sp24 = &sp0C->unk68.as_four_players;
-    for (r8 = 0, sp1C = 0, sp20 = 0; r8 < sp0C->unk9; r8++)
+    union BerryCrushGame_68 * sp24 = &sp0C->unk68;
+    s32 sp30;
+
+    for (; r8 < sp0C->unk9; r8++)
     {
-        // sp10<<2 --> sp28
-        // &sp0C->unk138.unk82 --> sp2C
         DynamicPlaceholderTextUtil_Reset();
         switch (sp10)
         {
         case 0:
-            sp1C = sp0C->unk68.as_five_players.unk1C[0].unk4.as_2d_bytes[sp10][r8];
-            if (r8 != 0 && sp24->unk00.unk0C[sp10][r8] != sp24->unk00.unk0C[sp10][r8 - 1])
+            sp1C = sp24->as_five_players.unk1C[0].unk4.as_2d_bytes[0][r8];
```

Comment:
does this not match with sp10 as the index in place of 0?

Reply:
yes. updated